### DISCUSSION TITLE
Session.exec does not pass mypy type-checking when passing delete statement

### DISCUSSION THREAD

**Discussion by jbbakst** [fastapi/sqlmodel#831] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional

from sqlmodel import Field, Session, SQLModel, col, delete

class Hero(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    name: str
    secret_name: str
    age: Optional[int] = None

def delete_heroes(ids: list[int], session: Session):
    session.exec(
        delete(Hero).where(
            col(Hero.id).in_(ids)
        )
    )
```

### Description

* Create a Hero model
* write a function that bulk deletes multiple heroes (like in the example code)
* run mypy

Expected result: the code passes type-checking

Actual result: mypy fails with the following error:
```
No overload variant of "exec" of "Session" matches argument type "Delete"
```

Using `session.execute` instead of `session.exec` passes type-checking, but that function is deprecated.

### Operating System

macOS

### SQLModel Version

0.0.14

### Python Version

3.9.18

**Comment by joachimhuet** [comment]
I think we should add an overload to the `exec` method to still have the possibility of passing an `_Executable` statement:

```
    @overload
    def exec(
        self,
        statement: _Executable,
        *,
        params: Optional[Union[Mapping[str, Any], Sequence[Mapping[str, Any]]]] = None,
        execution_options: Mapping[str, Any] = util.EMPTY_DICT,
        bind_arguments: Optional[Dict[str, Any]] = None,
        _parent_execute_state: Optional[Any] = None,
        _add_event: Optional[Any] = None,
    ) -> TupleResult[_TSelectParam]:
        ...

```

**Comment by shaleh** [comment]
Update.

Executable appears to be present. However, for some reason Delete is not seen as Executable.

```
        statement: Union[
            Select[_TSelectParam],
            SelectOfScalar[_TSelectParam],
            Executable[_TSelectParam],
        ],
```

I see that as of SQLModel 0.0.24.

Currently I am working around this with `# type: ignore[call-overload]`

**Comment by YuriiMotov** [comment]
There is an issue to track this: https://github.com/fastapi/sqlmodel/issues/909
And PR: https://github.com/fastapi/sqlmodel/pull/1342
